fe/fe-core/src/main/java/com/starrocks/catalog/RangePartitionInfo.java

```diff
@@ -102,6 +102,14 @@ public RangePartitionInfo(List<Column> partitionColumns) {
         this.isMultiColumnPartition = partitionColumns.size() > 1;
     }
 
+    public RangePartitionInfo(RangePartitionInfo other) {
+        super(PartitionType.RANGE);
```

Comment:
```suggestion
        super(other.getType());
```
The automatic partition will use PartitionType.EXPR_RANGE